dateutil/test/test_parser.py

```diff
@@ -921,6 +921,10 @@ def test_QBCD(self):
         with self.assertRaises(ValueError):
             parse('19-9-42-14QBCD', fuzzy=False, ignore_bad_tzname=False)
 
+    def test_QBCD_legacy(self):
+        dt = parse('19-9-42-14QBCD', fuzzy=False)
+        assert dt == datetime(2042, 9, 19, 14)
```

Comment:
Huh?  The issue with #478 is that this should raise, _not_ return `datetime(2042, 9, 19, 14)`

Reply:
I added this test to show that the function is backwards compatible if you don't supply the new `ignore_bad_tzname` argument.